common/app/routes/challenges/redux/reducer.js

```diff
@@ -64,6 +65,16 @@ const initialState = {
 
 const mainReducer = handleActions(
   {
+    [types.isPassed]: state => {
+      let { tests } = state;
+    if (tests.length > 0 && tests.every(test => test.pass && !test.err)) {
```

Comment:
Since you are checking a boolean to set a boolean, this can be done directly to set the property.